### ISSUE TITLE
Hires Fix takes ages with Flux

### ISSUE THREAD

**Issue by Pantonia4** [lllyasviel/stable-diffusion-webui-forge#2095]
Hires fix runs at like 33.59s/it while normal generation runs at maybe 3s/it on my end. Is there anything that can be done about it?

**Comment by ostap667inbox** [comment]
Would you like to start by writing down what video card you have, how much RAM you have on your system and, most importantly, what is the base generation size and what upscaling is set in hires.fix?

**Comment by arkhub** [comment]
similar case with me. I have 8gb RTX 3070 running flux nf4 with 32 Gb RAM with base resolution 1152x896 and hires.fix takes long time (about 8 min). vram usage almost reach 8 gb though but no shared memory used.

**Comment by ostap667inbox** [comment]
Yes, It is advisable to keep the system resources monitor window open while creating the image. In most cases, except when there is a real problem, all the “Why is it so slow?” questions will go away and you won't need to write the same questions to GitHub.

> vram usage almost reach 8 gb though but no shared memory used

Wrong. Forge makes heavy use of RAM and VRAM, and in some situations you may end up with a RAM deficit and shared memory may not even be utilized. Look into memory settings for FLUX, lllyasviel [made a separate page about it](https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/981). Seriously, you have huge weight checkpoints and text encoders. You don't think it's going to magically fit and spin around inside 8Gb of VRAM, do you?

8Gb VRAM + only 32Gb Ram for an 1152x896 image is already barely enough. And you're still trying to generate an image with four times the area! Once you run out of VRAM and RAM the system will start using a slow swap file. In the resource monitor you will immediately see constant disk access.

**Comment by NoMansPC** [comment]
Frankly, I'm surprised that Hiresfix already works with Flux in Forge. I'll give it a try.

**Comment by rockiecxh** [comment]
actually I tried the hresfix, but it doesn't work as expected, the upscaled images have gridlines.

**Comment by shaun-ba** [comment]
> actually I tried the hresfix, but it doesn't work as expected, the upscaled images have gridlines.

Can you explain and with parameters? Works fine for me.

**Comment by ostap667inbox** [comment]
> Can you explain and with parameters? Works fine for me.

Latent upscaler + denoising < 0.3 = grid

**Comment by NoMansPC** [comment]
> > Can you explain and with parameters? Works fine for me.
> > I'll take a guess:
> > Latent upscaler + denoising < 0.3 = grid

Same for me. Upscaling just doesn't work with Flux in Forge. It's been like this since it was launched.

**Comment by shaun-ba** [comment]
Can someone attach a screenshot showing what they mean by grid lines, because my upscales look prefect to me

**Comment by c4801725870** [comment]
> > Can you explain and with parameters? Works fine for me.
> > I'll take a guess:
> > Latent upscaler + denoising < 0.3 = grid

This is just user error, Latent need denoise larger than 0.5. I'm surprised it worked at all. Try reading about the Latent method

**Comment by NoMansPC** [comment]
> > > Can you explain and with parameters? Works fine for me.
> > > I'll take a guess:
> > > Latent upscaler + denoising < 0.3 = grid
> 
> This is just user error, Latent need denoise larger than 0.5. I'm surprised it worked at all. Try reading about the Latent method

This is correct. I kept it at 0.7 denoise and the end result looked great. I thought that it would change too much from the initial image.

**Comment by c4801725870** [comment]
> > > > Can you explain and with parameters? Works fine for me.
> > > > I'll take a guess:
> > > > Latent upscaler + denoising < 0.3 = grid
> > 
> > 
> > This is just user error, Latent need denoise larger than 0.5. I'm surprised it worked at all. Try reading about the Latent method
> 
> This is correct. I kept it at 0.7 denoise and the end result looked great. I thought that it would change too much from the initial image.

Yea you gotta think in terms of the maths, the image is totally different if details are added. To our brains it doesn't seem much different but mathematically it is. 0.7 on a human will give it freedom to create detailed eyelashes, hairs on the forearm skin, texture of skin pores, all these things seem like minor details but it is hugely different vs the base image. Most people think it means changes to the scenery or pose or something that matters to us, but that is really dictated by a solid prompt and appropriate Distilled CFG and CFG values, along with playing with LORA weights when twinning or mutations occur.
